sale_stock_cancel_restriction/models/sale_order.py

```diff
@@ -0,0 +1,12 @@
+# Copyright 2021 Tecnativa - Ernesto Tejeda
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_cancel(self):
+        self.mapped("picking_ids").filtered(lambda r: r.state == "done").action_cancel()
```

Comment:
Put a docstring explaining the whys:
```suggestion
        """Force to call the cancel method on done picking for having the expected error,
        as Odoo has now filter out such pickings from the cancel operation.
        """
        self.mapped("picking_ids").filtered(lambda r: r.state == "done").action_cancel()
```